java/flight/flight-sql-jdbc-core/src/main/java/org/apache/arrow/driver/jdbc/converter/AvaticaParameterConverter.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.driver.jdbc.converter;
+
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.calcite.avatica.AvaticaParameter;
+import org.apache.calcite.avatica.remote.TypedValue;
+
+/**
+ * Interface for a class in charge of converting between AvaticaParameters and TypedValues and
+ * Arrow.
+ */
+public interface AvaticaParameterConverter {
+
+  /**
+   * Bind a TypedValue to a FieldVector at the given index.
+   *
+   * @param vector FieldVector that the parameter should be bound to.
+   * @param typedValue TypedValue to bind as a parameter.
+   * @param index Vector index that the TypedValue should be bound to.
```

Comment:
0-index, right? (I ask because JDBC is often 1-indexed)

Reply:
Correct! that would be the index on the Arrow vector. I'll add to the docstring just to be more clear